hack/deploy/configure-values.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script interrogates the AKS cluster and Azure resources to generate 
+# the karpenter-values.yaml file using the karpenter-values-template.yaml file as a template.
+
+# Check the cluster name and resource group are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <cluster-name> <resource-group>"
+    exit 1
+fi
+
+echo "Configuring karpenter-values.yaml for cluster $1 in resource group $2 ..."
+
+AZURE_CLUSTER_NAME=$1
+AZURE_RESOURCE_GROUP=$2
+
+AZURE_LOCATION=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".location")
+AZURE_RESOURCE_GROUP_MC=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".nodeResourceGroup")
+
+KARPENTER_SERVICE_ACCOUNT_NAME=karpenter-sa
+AZURE_KARPENTER_USER_ASSIGNED_IDENTITY_NAME=karpentermsi
+
+AZURE_CLIENT_ID=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".identityProfile.kubeletidentity.clientId")
+CLUSTER_ENDPOINT=$(kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}')
+
+TOKEN_SECRET_NAME=$(kubectl get -n kube-system secrets --field-selector=type=bootstrap.kubernetes.io/token -o jsonpath='{.items[0].metadata.name}')
+TOKEN_ID=$(kubectl get -n kube-system secret "$TOKEN_SECRET_NAME" -o jsonpath='{.data.token-id}' | base64 -d)
+TOKEN_SECRET=$(kubectl get -n kube-system secret "$TOKEN_SECRET_NAME" -o jsonpath='{.data.token-secret}' | base64 -d)
+BOOTSTRAP_TOKEN=$TOKEN_ID.$TOKEN_SECRET
+
+SSH_PUBLIC_KEY="$(cat ~/.ssh/id_rsa.pub) azureuser"
+AZURE_VNET_NAME=$(az network vnet list --resource-group "$AZURE_RESOURCE_GROUP_MC" |  jq -r ".[0].name")
```

Comment:
nit: again is there a way to reuse `az network vnet list --resource-group "$AZURE_RESOURCE_GROUP_MC"`

However, this is pretty simplistic, and might not be worth it.